### ISSUE TITLE
[🐛] iOS launchCamera doesn't return permission error code , just opens camera with black background

### ISSUE THREAD

**Issue by gugakatsi** [react-native-image-picker/react-native-image-picker#2024]
### Description

Explain the bug in as much detail as possible

After rejecting camera permission on iOS, it doesn't return errorCode , just opens camera with black background and 'cancel' option, tapping on Snap ( circle ) button doesn't do anything.

### How to repeat issue and example
* install rn-image-picker 4.10.0
* launch physical ios device 
* try opening camera 
* reject camera permission
* try opening camera again

- step to reproduce the issue
- ...

### Solution

What needs to be done to address this issue?

### Additional Information

- Image Picker version: 4.10.0
- React Native version: 0.66.1
- Platform: iOS
- Development Operating System: MacOS 
- Dev tools: Xcode

**Comment by Rookie088** [comment]
same here

**Comment by umairalirehan** [comment]
its due to back camera side. you just turn on front camera for taking selfie it will work. hopefully it will help you

**Comment by gugakatsi** [comment]
@umairalirehan you mean it does return error code , but while you try to snap it with front (selfie) camera ?

**Comment by umairalirehan** [comment]
No. I am saying if you are facing Black screen when user open camera. Then
by default back camera turn on. If you face this issue you just turn on
front camera. Because user can just take picture from front camera in this s
Package I think and maybe it's a bug of this package.

**Comment by sn4f** [comment]
same here

**Comment by richard-access** [comment]
Also have this issue. Based on the docs, I expect this to return the permissions related error code.

**Comment by Reisclef** [comment]
Not sure how often the Pull Requests are looked at for this library, but I've shifted some of the code around in the file and it seems to no longer open the camera when permission is denied.

In order to accomplish this, I refactored some camera-specific logic out of the launchImagePicker function.

Would be good to know if this fixes the issue for others in this thread.

**Comment by ivosw** [comment]
Thanks. I made a patch based on your code changes and it works as expected.

**Comment by remod** [comment]
For me it works if I additinally grant permissions to access the image galery.
